diff --git a/node_modules/@imkey/web3-provider/dist/imkey-web3-provider.umd.js b/node_modules/@imkey/web3-provider/dist/imkey-web3-provider.umd.js
index 9efd318..822b6da 100644
--- a/node_modules/@imkey/web3-provider/dist/imkey-web3-provider.umd.js
+++ b/node_modules/@imkey/web3-provider/dist/imkey-web3-provider.umd.js
@@ -62978,7 +62978,11 @@
         close() {
             return __awaiter(this, void 0, void 0, function* () {
                 if (this.transport) {
-                    yield this.transport.close();
+                    try {
+                      yield this.transport.close();
+                    } catch (e) {
+                      console.log(e);
+                    }
                 }
                 this.transport = null;
                 this.eth = null;
