diff --git a/node_modules/@ledgerhq/hw-app-eth/lib-es/services/ledger/contracts.js b/node_modules/@ledgerhq/hw-app-eth/lib-es/services/ledger/contracts.js
index c3deee8..2556634 100644
--- a/node_modules/@ledgerhq/hw-app-eth/lib-es/services/ledger/contracts.js
+++ b/node_modules/@ledgerhq/hw-app-eth/lib-es/services/ledger/contracts.js
@@ -10,7 +10,7 @@ export const loadInfosForContractMethod = async (contractAddress, selector, chai
     if (pluginBaseURL) {
         const url = `${pluginBaseURL}/plugins/ethereum.json`;
         data = await axios
-            .get(`${pluginBaseURL}/plugins/ethereum.json`)
+            .get(`${pluginBaseURL}/plugins/ethereum.json`, { timeout: 5000 })
             .then(r => r.data)
             .catch(e => {
             log("error", "could not fetch from " + url + ": " + String(e));
diff --git a/node_modules/@ledgerhq/hw-app-eth/lib-es/services/ledger/erc20.js b/node_modules/@ledgerhq/hw-app-eth/lib-es/services/ledger/erc20.js
index 0374c81..3050694 100644
--- a/node_modules/@ledgerhq/hw-app-eth/lib-es/services/ledger/erc20.js
+++ b/node_modules/@ledgerhq/hw-app-eth/lib-es/services/ledger/erc20.js
@@ -12,7 +12,7 @@ export const findERC20SignaturesInfo = async (userLoadConfig, chainId) => {
         return null;
     const url = `${cryptoassetsBaseURL}/evm/${chainId}/erc20-signatures.json`;
     const blob = await axios
-        .get(url)
+        .get(url, { timeout: 5000 })
         .then(({ data }) => {
         if (!data || typeof data !== "string") {
             throw new Error(`ERC20 signatures for chainId ${chainId} file is malformed ${url}`);
