diff --git a/node_modules/@metamask/browser-passworder/dist/index.js b/node_modules/@metamask/browser-passworder/dist/index.js
index 6c719e7..f788404 100644
--- a/node_modules/@metamask/browser-passworder/dist/index.js
+++ b/node_modules/@metamask/browser-passworder/dist/index.js
@@ -135,7 +135,7 @@ exports.decryptWithKey = decryptWithKey;
  * @returns A CryptoKey.
  */
 async function importKey(keyString) {
-    const key = await window.crypto.subtle.importKey(EXPORT_FORMAT, JSON.parse(keyString), DERIVED_KEY_FORMAT, true, ['encrypt', 'decrypt']);
+    const key = await crypto.subtle.importKey(EXPORT_FORMAT, JSON.parse(keyString), DERIVED_KEY_FORMAT, true, ['encrypt', 'decrypt']);
     return key;
 }
 exports.importKey = importKey;
@@ -147,7 +147,7 @@ exports.importKey = importKey;
  * @returns A key string.
  */
 async function exportKey(key) {
-    const exportedKey = await window.crypto.subtle.exportKey(EXPORT_FORMAT, key);
+    const exportedKey = await crypto.subtle.exportKey(EXPORT_FORMAT, key);
     return JSON.stringify(exportedKey);
 }
 exports.exportKey = exportKey;
