diff --git a/node_modules/@rabby-wallet/eth-walletconnect-keyring/dist/v2sdk.js b/node_modules/@rabby-wallet/eth-walletconnect-keyring/dist/v2sdk.js
index 0000000..0000000 100644
--- a/node_modules/@rabby-wallet/eth-walletconnect-keyring/dist/v2sdk.js
+++ b/node_modules/@rabby-wallet/eth-walletconnect-keyring/dist/v2sdk.js
@@ -138,7 +138,7 @@ class WalletConnectKeyringV2 extends WalletConnectKeyringBase {
             };
             const listenerJwtError = () => {
                 var _a;
-                (_a = this.client) === null || _a === void 0 ? void 0 : _a.core.relayer.provider.once('error', (e) => __awaiter(this, void 0, void 0, function* () {
+                (_a = this.client) === null || _a === void 0 ? void 0 : _a.core.relayer.once('error', (e) => __awaiter(this, void 0, void 0, function* () {
                     var _b;
                     // error code 3000 meaning the jwt token is expired, need to re-init the client
                     // only appear in connect method
