diff --git a/node_modules/@rabby-wallet/rabby-api/dist/types.d.ts b/node_modules/@rabby-wallet/rabby-api/dist/types.d.ts
index c0810ab..ecae766 100644
--- a/node_modules/@rabby-wallet/rabby-api/dist/types.d.ts
+++ b/node_modules/@rabby-wallet/rabby-api/dist/types.d.ts
@@ -207,6 +207,11 @@ export interface Spender {
     is_abandoned: boolean;
     risk_alert: string;
     risk_level: string;
+    exposure_nft_usd_value?: number | null;
+    exposure_usd_value?: number | null;
+    approve_user_count?: number | null;
+    revoke_user_count?: number | null;
+    last_approve_at?: number | null;
 }
 export interface AssetItem {
     id: string;
