diff --git a/node_modules/@trezor/connect-web/lib/webextension/trezor-usb-permissions.js b/node_modules/@trezor/connect-web/lib/webextension/trezor-usb-permissions.js
index 6d84980..dbe16e6 100644
--- a/node_modules/@trezor/connect-web/lib/webextension/trezor-usb-permissions.js
+++ b/node_modules/@trezor/connect-web/lib/webextension/trezor-usb-permissions.js
@@ -12,15 +12,15 @@ const switchToPopupTab = event => {
             currentWindow: true,
             active: true,
         }, current => {
-            if (current.length < 0)
+            if (current[0]?.id == null)
                 return;
             chrome.tabs.remove(current[0].id);
         });
     }
     chrome.tabs.query({
-        url: `${url}popup.html`,
+        url: `${url}popup.html*`,
     }, tabs => {
-        if (tabs.length < 0)
+        if (tabs[0]?.id == null)
             return;
         chrome.tabs.update(tabs[0].id, { active: true });
     });
